v2/src/lib.rs

```diff
@@ -475,21 +466,25 @@ impl Storage for InfluxDbStorage {
 
         // Note: assume that uhlc timestamp was generated by a clock using UNIX_EPOCH (that's the case by default)
         // encode the value as a string to be stored in InfluxDB, converting to base64 if the buffer is not a UTF-8 string
-        let (base64, strvalue) = match String::from_utf8(value.payload.contiguous().into_owned()) {
-            Ok(s) => (false, s),
-            Err(err) => (true, b64_std_engine.encode(err.into_bytes())),
-        };
+        let (base64, strvalue) =
+            match String::from_utf8(ZBuf::from(value.payload()).contiguous().into_owned()) {
+                Ok(s) => (false, s),
+                Err(err) => (true, b64_std_engine.encode(err.into_bytes())),
+            };
 
         let db = get_db_name(self.config.clone())?;
 
         // Note: tags are stored as strings in InfluxDB, while fileds are typed.
         // For simpler/faster deserialization, we store encoding, timestamp and base64 as fields.
         // while the kind is stored as a tag to be indexed by InfluxDB and have faster queries on it.
+        let encoding_string_rep = value.encoding().clone().to_string(); // TODO: This i am not entirely sure about
+        let encoding: Encoding = (value.encoding().clone()).into();
+
         let zenoh_point = vec![DataPoint::builder(measurement.clone())
             .tag("kind", "PUT")
             .field("timestamp", timestamp.to_string())
-            .field("encoding_prefix", u8::from(*value.encoding.prefix()) as i64) //should be u8 but not supported in v2.x so using a workaround
-            .field("encoding_suffix", value.encoding.suffix())
+            .field("encoding_prefix", u16::from(encoding.id()) as i64) //should be u8 but not supported in v2.x so using a workaround
+            .field("encoding_suffix", encoding_string_rep) // TODO: This i am not entirely sure about
```

Comment:
replace comment to like "TODO: left for compatibility, consider replacing to single "encoding" field